osquery/tables/system/windows/wmi_bios_info.cpp

```diff
@@ -155,52 +161,329 @@ Row getDellBiosInfo(const WmiResultItem& item) {
+                                                            "SMBIOSPresent"};
+
+  static const std::vector<std::string> kDateKeyNameList{"InstallDate",
+                                                         "ReleaseDate"};
+
+  auto exp_wmi_request = WmiRequest::CreateWmiRequest(kWin32BiosQuery);
+  if (exp_wmi_request.isError()) {
+    const auto& error = exp_wmi_request.getError();
+
+    LOG(ERROR) << "The following WMI query could not be constructed: "
+               << kWin32BiosQuery << ". " << error.getMessage();
+
     return {};
   }
-  const auto& wmiComputerSystemResults = wmiComputerSystemReq->results();
 
-  std::string manufacturer;
-  wmiComputerSystemResults[0].GetString("Manufacturer", manufacturer);
-  manufacturer = getManufacturer(manufacturer);
+  auto wmi_request = exp_wmi_request.take();
+  if (!wmi_request.getStatus().ok()) {
+    LOG(ERROR) << "The following WMI query has failed: " << kWin32BiosQuery;
+    return {};
+  }
 
-  auto it = kQueryMap.find(manufacturer);
-  if (it == kQueryMap.end()) {
-    LOG(INFO) << "Vendor \"" << manufacturer << "\" is currently not supported";
+  const auto& wmi_item_list = wmi_request.results();
+  if (wmi_item_list.empty()) {
+    LOG(ERROR) << "The following WMI query did not return any item: "
+               << kWin32BiosQuery;
     return {};
   }
 
-  QueryData results;
-  const auto wmiBiosReq = WmiRequest::CreateWmiRequest(std::get<0>(it->second),
-                                                       std::get<1>(it->second));
-  if (!wmiBiosReq) {
-    LOG(WARNING) << wmiBiosReq.getError().getMessage();
-    return results;
+  if (wmi_item_list.size() != 1) {
+    LOG(ERROR)
+        << "The following WMI query returned an unexpected number of items: "
+        << kWin32BiosQuery;
+  }
+
+  const auto& wmi_item = wmi_item_list.back();
+
+  QueryData row_list;
+  auto L_generateRow = [&row_list](std::string key, std::string value) {
+    Row row{};
+    row["name"] = std::move(key);
+    row["value"] = std::move(value);
+
+    row_list.push_back(std::move(row));
+  };
+
+  for (const auto& key_name : kStringKeyNameList) {
+    std::string value;
+    auto status = wmi_item.GetString(key_name, value);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+
+      value.clear();
+    }
+
+    L_generateRow(key_name, std::move(value));
+  }
+
+  for (const auto& key_name : kStringVectorKeyNameList) {
+    std::vector<std::string> value_list;
+    auto status = wmi_item.GetVectorOfStrings(key_name, value_list);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+
+      value_list.clear();
+    }
+
+    std::string buffer = "{";
+    for (const auto& value : value_list) {
+      if (buffer.size() > 1) {
+        buffer += ", ";
+      }
+
+      buffer += value;
+    }
+
+    buffer += "}";
```

Comment:
Not strictly EAV style, but maybe good enough. (And this is table is missing E, so it's harder)